nbextensions/usability/execute_time/ExecuteTime.js

```diff
@@ -1,182 +1,240 @@
 
 //  Copyright (C) 2014  Jean-Christophe Jaskula
+//                2015  joshuacookebarnes@gmail.com
 //
 //  Distributed under the terms of the BSD License.
-//----------------------------------------------------------------------------
+// ---------------------------------------------------------------------------
 //
-// Execute timings - display when a cell has been executed lastly and how long it took
-// A double click on the box makes it disappear
-define(["require", "jquery", "base/js/namespace", "base/js/events"], function (require, $, IPython, events) {
-    "use strict";
-
-    var firstExecTime=null;
-    var execCells=[];
-    var toggle_all = null;
-
-    var month_names = ["Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"];
-    var day_names = ["Sun","Mon","Tue","Wed","Thu","Fri","Sat"];
-
-    var patchCodecellExecute = function() {
-        console.log('patching codecell to trigger ExecuteCell.ExecuteTime');
-        IPython.CodeCell.prototype.old_execute = IPython.CodeCell.prototype.execute;
-
-        IPython.CodeCell.prototype.execute = function () {
-            this.old_execute(arguments);
-            events.trigger('ExecuteCell.ExecuteTime');
+// Execution timings:
+// display when a cell was last executed, and how long it took to run
+// A double click on the timing box makes it disappear
+
+define([
+    'require',
+    'jquery',
+    'moment',
+    'base/js/namespace',
+    'base/js/events',
+    'notebook/js/codecell'
+], function (
+    require,
+    $,
+    moment,
+    Jupyter,
+    events,
+    codecell
+) {
+    'use strict';
+
+    var CodeCell = codecell.CodeCell;
+
+    function patch_CodeCell_get_callbacks () {
+        console.log('[ExecuteTime] patching CodeCell.prototype.get_callbacks to insert an ExecuteTime shell.reply callback');
+        var old_get_callbacks = CodeCell.prototype.get_callbacks;
+        CodeCell.prototype.get_callbacks = function () {
+            var callbacks = old_get_callbacks.apply(this, arguments);
+
+            var cell = this;
+            var prev_reply_callback = callbacks.shell.reply;
+            callbacks.shell.reply = function (msg) {
+                if (msg.msg_type === 'execute_reply') {
+                    $.extend(true, cell.metadata, {
+                        ExecuteTime: {
+                            start_time: msg.metadata.started,
+                            end_time: msg.header.date
+                        }
+                    });
+                    var timing_area = update_timing_area(cell);
+                    if ($.ui !== undefined) {
+                        timing_area.stop(true, true).show(0).effect('highlight', {color: '#0F0'});
```

Comment:
A very minor nitpick, but would you mind changing that color to `#0B0`? The pure green is disturbingly bright, and I guess the original code used `#b00000` for that same reason.